docs/architecture.md

```diff
@@ -0,0 +1,25 @@
+## Architecture
```

Comment:
Can we replace it with https://github.com/Cloud-CV/EvalAI/wiki/Architecture? Minor changes done.